projects/scion/microfrontend-platform/src/lib/host/message-broker/message-broker.ts

```diff
@@ -65,11 +66,12 @@ export class MessageBroker implements PreDestroy {
     this._manifestRegistry = Beans.get(ManifestRegistry);
 
     // Construct a stream of messages sent by clients.
-    this._clientRequests$ = fromEvent<MessageEvent>(window, 'message')
+    this._clientMessage$ = fromEvent<MessageEvent>(window, 'message')
       .pipe(
-        filterMessage(MessagingTransport.ClientToBroker),
+        filterByTransport(MessagingTransport.ClientToBroker),
+        filterByChannel(MessagingChannel.Intent, MessagingChannel.Topic, MessagingChannel.TopicSubscribe, MessagingChannel.TopicUnsubscribe),
```

Comment:
What's the point of this filter, since we're filtering again further down the stream? Performance?

Reply:
Do you mean the channel filter? It's necessary because we are installing the `checkOriginTrusted` operator later down. We can verify the origin only for listed channels, but, for example, not for the `ClientConnect` channel. For that reason, we explicitly list expected channels.